File: packages/extract/test/fixtures/polyglot/kotlin/Job.kt
package app

enum class State { IDLE, RUNNING, DONE, FAILED, RETRYING }

object Registry {
    val jobs = mutableListOf<Job>()
}

class Job {
    private var state: State = State.IDLE

    fun start() {
        if (state != State.IDLE) return
        state = State.RUNNING
    }

    fun finish() {
        if (state == State.RUNNING) {
            state = State.DONE
        }
    }

    fun fail() {
        state = State.FAILED
    }

    fun describe(): String = when (state) {
        State.IDLE -> "waiting"
        State.RUNNING -> "busy"
        else -> "other"
    }
}

File: packages/extract/test/fixtures/polyglot/scala/Job.scala
package app

sealed trait State
case object Idle extends State
case object Running extends State
case object Done extends State
case object Failed extends State
case object Retrying extends State

class Job {
  private var state: State = Idle

  def start(): Unit = {
    if (state != Idle) return
    state = Running
  }

  def finish(): Unit = {
    if (state == Running) state = Done
  }

  def fail(): Unit = {
    state = Failed
  }
}
